lib/matplotlib/legend.py

```diff
@@ -974,3 +1105,145 @@ def draggable(self, state=None, use_blit=False, update="loc"):
             self._draggable = None
 
         return self._draggable
+
+
+# Helper functions to parse legend arguments for both `figure.legend` and
+# `axes.legend`:
+def _get_legend_handles(axs, legend_handler_map=None):
+    """
+    Return a generator of artists that can be used as handles in
+    a legend.
+
+    """
+    handles_original = []
+    for ax in axs:
+        handles_original += (ax.lines + ax.patches +
+                        ax.collections + ax.containers)
+        # support parasite axes:
+        if hasattr(ax, 'parasites'):
+            for axx in ax.parasites:
+                handles_original += (axx.lines + axx.patches +
+                        axx.collections + axx.containers)
+
+    handler_map = Legend.get_default_handler_map()
+
+    if legend_handler_map is not None:
+        handler_map = handler_map.copy()
+        handler_map.update(legend_handler_map)
+
+    has_handler = Legend.get_legend_handler
+
+    for handle in handles_original:
+        label = handle.get_label()
+        if label != '_nolegend_' and has_handler(handler_map, handle):
+            yield handle
+
+
+def _get_legend_handles_labels(axs, legend_handler_map=None):
+    """
+    Return handles and labels for legend
+
+    ``ax.legend()`` is equivalent to ::
+
+      h, l = ax.get_legend_handles_labels()
```

Comment:
Should this be `ax._get_legend_handles_labels()` (missing underscrore)?

Reply:
Not really - there is a `ax.get_legend_handles_labels()`  But it doesn't need to be referenced here!